auth4genai/snippets/get-started/react-spa-vercel-ai/call-others-api.mdx

```diff
@@ -0,0 +1,635 @@
+- Note down the `client_id` and `client_secret` for your environment variables after running curl successfully.
+- This client enables federated token exchange to get access tokens for external connections (e.g., Google Calendar)
+
+#### 4. Configure Social Connection
+
+- Configure a Google Social Connection in Auth0
+- Make sure to enable all `Calendar` scopes from the Permissions options
+- Make sure to enable Token Vault for the Connection under the Advanced Settings
+- Make sure to enable the connection for your SPA Application and the Resource Server Client
+- Test the connection in Auth0 "Try Connection" screen
+
+### Prepare React SPA + Hono API
+
+**Recommended**: To use a starter template, clone the [Auth0 AI samples](https://github.com/auth0-lab/auth0-ai-js) repository:
+
+```bash wrap lines
+git clone https://github.com/auth0-lab/auth0-ai-js.git
+cd auth0-ai-js/examples/calling-apis/spa-with-backend-api/react-hono-ai-sdk
+```
+
+### Install dependencies
+
+In the root directory of your project, note some of the following dependencies:
+
+**Client dependencies:**
+- `@auth0/auth0-spa-js`: Auth0 SPA SDK for client-side authentication
+- `@auth0/ai-vercel`: [Auth0 AI SDK for Vercel AI](https://github.com/auth0-lab/auth0-ai-js/tree/main/packages/ai-vercel) built for GenAI applications
+- `ai`: Core [Vercel AI SDK](https://sdk.vercel.ai/docs) module
+
+**Server dependencies:**
+- `@hono/node-server`: Node.js server adapter for Hono
+- `hono`: Lightweight web framework
+- `ai`: Core [Vercel AI SDK](https://sdk.vercel.ai/docs) module
+- `@ai-sdk/openai`: [OpenAI](https://sdk.vercel.ai/providers/ai-sdk-providers/openai) provider
+- `googleapis`: Node.js client for Google APIs
+- `jose`: JavaScript Object Signing and Encryption library for JWT verification
+
+```bash wrap lines
+# Install client & server dependencies from the root directory of the project.
+# All workspace dependencies will be installed automatically.
+npm install
+```
+
+### Update the environment files
+
+#### Client (.env)
+```bash .env wrap lines
+VITE_AUTH0_DOMAIN=your-auth0-domain
+VITE_AUTH0_CLIENT_ID=your-spa-client-id
+VITE_AUTH0_AUDIENCE=your-api-identifier
+VITE_API_URL=http://localhost:3001
+```
+
+#### Server (.env)
+```bash .env wrap lines
+AUTH0_DOMAIN=your-auth0-domain
+AUTH0_AUDIENCE=your-api-identifier
+AUTH0_CLIENT_ID=your-resource-server-client-id
+AUTH0_CLIENT_SECRET=your-resource-server-client-secret
+OPENAI_API_KEY=your-openai-api-key
+PORT=3001
+```
+
+### Set up federated connection with access tokens
+
+#### Configure the SPA for step-up authorization
+
+Unlike the Next.js example which uses refresh tokens, this React SPA approach uses **access tokens** for federated token exchange. The SPA handles step-up authorization using Auth0 SPA SDK's `loginWithPopup()` method to display the consent screen, and allow the user to grant additional permissions.
```

Comment:
Mention of Next.js example is ambiguous, maybe a link to it will help

Reply:
5398b4f24834da42e367c88b55028bf0f2bce4f1